twitter/tweets_test.go

```diff
@@ -0,0 +1,71 @@
+package twitter
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+func TestTweetSerialization(t *testing.T) {
+	tweet := loadTestTweet()
+	if tweet.User.ID != 971761 {
+		t.Fail()
+	}
+
+}
+
+func TestLocationSerialization(t *testing.T) {
+	tweet := loadTestLocationTweet()
+	if tweet.User.ID != 971761 {
+		t.Fail()
+	}
+	place := tweet.Place
+	if place == nil {
+		t.Fail()
+	}
+
+	if len(place.BoundingBox.Coordinates) == 0 {
+		t.Fail()
+	}
+	if place.BoundingBox.Type != "Polygon" {
+		t.Fail()
+	}
+	if place.Country != "United States" {
+		t.Fail()
+	}
+	if place.CountryCode != "US" {
+		t.Fail()
+	}
+}
+
+func loadTestTweet() *Tweet {
+	f, e := ioutil.ReadFile("./test_data/tweet.json")
+	if e != nil {
+		fmt.Println("File error: ", e)
+		os.Exit(1)
+	}
+	var twt Tweet
+	err := json.Unmarshal(f, &twt)
+	if err != nil {
+		fmt.Println("There was an error unmarshalling", err)
+		os.Exit(1)
+	}
+	return &twt
+}
+
+func loadTestLocationTweet() *Tweet {
+	f, e := ioutil.ReadFile("./test_data/location_tweet.json")
```

Comment:
Dude I appreciate it I wrote my first few lines of go two weeks ago haha - love the community - thanks for the incredibly useful advice 





On Thu, Jun 16, 2016 at 7:10 PM +1000, "Dmitri Shuralyov" <notifications@github.com> wrote:












In twitter/tweets_test.go:
> +	f, e := ioutil.ReadFile("./test_data/tweet.json")
> +	if e != nil {
> +		fmt.Println("File error: ", e)
> +		os.Exit(1)
> +	}
> +	var twt Tweet
> +	err := json.Unmarshal(f, &twt)
> +	if err != nil {
> +		fmt.Println("There was an error unmarshalling", err)
> +		os.Exit(1)
> +	}
> +	return &twt
> +}
> +
> +func loadTestLocationTweet() *Tweet {
> +	f, e := ioutil.ReadFile("./test_data/location_tweet.json")



No problem. :)



Keep in mind I'm just a random person, not a developer/maintainer of this project. I've also not reviewed the rest of the PR in full, I just wanted to comment on what I was able to spot when looking over it quickly. :)



—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.